src/test/java/com/frameworkium/integration/ai/capture/api/dto/executions/BaseDTO.java

```diff
@@ -0,0 +1,46 @@
+package com.frameworkium.integration.ai.capture.api.dto.executions;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+
+/**
+ * Base Data transfer object for all data objects to extend from
+ */
+public class BaseDTO<T extends BaseDTO> {
+
+    @Override
+    public String toString() {
+        return ToStringBuilder.reflectionToString(this);
+    }
+
+    @SuppressWarnings("EqualsWhichDoesntCheckParameterClass")
+    public boolean equals(Object obj) {
+        return EqualsBuilder.reflectionEquals(this, obj);
+    }
+
+    /**
+     * This method makes a clone of any Java object it is given.
+     * Use to clone an object you receive as a response before
+     * modifying instead of directly modifying the response.
+     */
+    public static Object clone(Object object) {
+        try {
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            ObjectOutputStream oos = new ObjectOutputStream(baos);
+            oos.writeObject(object);
+            ByteArrayInputStream bais = new ByteArrayInputStream(baos.toByteArray());
+            ObjectInputStream ois = new ObjectInputStream(bais);
+            return ois.readObject();
+        }
+        catch (Exception e) {
+            e.printStackTrace();
```

Comment:
We should probably throw rather than simply print. e.g. `throw new RuntimException(e);`. If this fails then there's no way the test could continue, and it's best to error at the point of error rather than later in a test when comparing something to the `null` returned here.